pkg/gpu/stream.go

```diff
@@ -197,11 +223,12 @@ func (sh *StreamHandler) handleKernelLaunch(event *gpuebpf.CudaKernelLaunch) {
 
 // trySendToChannel attempts to send an item to a channel in a non-blocking way, if the channel is full
 // it will increment the rejectedSpans telemetry counter
-func trySendSpan[T any](sh *StreamHandler, ch chan T, item T) {
+func trySendSpan[T any](sh *StreamHandler, ch chan *T, item *T, pool ddsync.Pool[T]) {
```

Comment:
i don't think trySendSpan should be in charge of managing the pool, maybe better have some return value and leave that logic to the caller?